### PR TITLE
code cleanup - common folder

### PR DESCRIPTION
**Pull request by tal-nc**
there are way too many files directly in corona_hakab_model. i think we should organize it.
started by adding 'common' folder for commonly used code.
also split a few multi-objects files into different files (no reason for Circle to be in agent.py).

### CODE REVIEW

File: src/corona_hakab_model/common/agent.py [hunk 1]
```diff
@@ -6,12 +6,12 @@
 import pandas as pd
 from numpy import nan
 
-from generation.circles import SocialCircleConstraint
+from common.social_circle import SocialCircleConstraint
```
Comment (ghost) [inline src/corona_hakab_model/common/agent.py]:
```suggestion
from .social_circle import SocialCircleConstraint
```

**Comment by ghost** [comment]
Maybe move detection_model/detection_testing_types.py too?

File: src/corona_hakab_model/common/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+__all__ = [
```
Comment (ghost) [inline src/corona_hakab_model/common/__init__.py]: why do we need this?

**Review by ghost — CHANGES_REQUESTED** [review]
I really like this change, it will make the project look a lot better.
Just some small fixes and we'll merge it.

File: src/corona_hakab_model/common/agent.py (hunk 1 above)
Comment (tal-nc) [reply src/corona_hakab_model/common/agent.py]: done

File: src/corona_hakab_model/common/__init__.py (hunk 2 above)
Comment (tal-nc) [reply src/corona_hakab_model/common/__init__.py]: i though we might wanna use it some times, it allows us to do 
`from common import *`
however we are not using it now, so i will remove it

**Review by ghost — APPROVED** [review]
looks good, good job

File: src/corona_hakab_model/common/__init__.py (hunk 2 above)
Comment (ghost) [reply src/corona_hakab_model/common/__init__.py]: oh that's what it does, i didn't know

**Comment by ghost** [comment]
just need to fix a little conflict

**Pull request merged**